framework/Metrics/metrics/StatsTestMetric.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Created on 2021 September 12
+
+@author: Robert Flanagan
+"""
+import numpy as np
+from scipy import stats
+
+from .MetricInterface import MetricInterface
+from Metrics.metrics import MetricUtilities
+from utils import InputData
+import StatsTestUtils as sU
+
+class StatsTestMetric(MetricInterface):
+  """
+    Metric to compare two datasets using statistical tests.
+  """
+  available_tests = {}
+  available_tests['full']['f_test'] = stats.f_oneway
+  available_tests['full']['chi_square'] = stats.chi_square
+  available_tests['full']['ks_test'] = stats.ks_2samp
+  available_tests['seg']['f_test'] = sU.compare_segments_f
+  available_tests['seg']['chi_square'] = sU.compare_segments_chi
+  available_tests['seg']['ks_test'] = sU.compare_segments_ks
+  available_tests['int']['f_test'] = sU.compare_interval_f
+  available_tests['int']['chi_square'] = sU.compare_interval_chi
+  available_tests['int']['ks_test'] = sU.compare_interval_ks
+  available_tests['seg_sum']['f_test'] = sU.compare_sums_f
+  available_tests['seg_sum']['chi_square'] = sU.compare_sums_chi
+  available_tests['seg_sum']['ks_test'] = sU.compare_sums_ks
```

Comment:
RAVEN naming conventions don't allow for `snake_case`, requiring all variables to be `camelCase`.